drivers/ethernet/eth_stm32_hal.c

```diff
@@ -957,10 +957,61 @@ void HAL_ETH_TxCpltCallback(ETH_HandleTypeDef *heth_handle)
 #if defined(CONFIG_ETH_STM32_HAL_API_V2)
 void HAL_ETH_ErrorCallback(ETH_HandleTypeDef *heth)
 {
-	/* Do nothing */
-	/* Do not log errors. If errors are reported du to high traffic,
+	/* Do not log errors. If errors are reported due to high traffic,
 	 * logging errors will only increase traffic issues
 	 */
+
+	uint32_t dma_error, mac_error;
+
+	__ASSERT_NO_MSG(heth != NULL);
+
+	const uint32_t error_code = HAL_ETH_GetError(heth);
+
+	struct eth_stm32_hal_dev_data *dev_data =
+		CONTAINER_OF(heth, struct eth_stm32_hal_dev_data, heth);
+
+	__ASSERT_NO_MSG(dev_data != NULL);
```

Comment:
I know this is already checked in other functions as well, but this is really redundant.
`dev_data` is assigned at build time, there is no reason to check it systematically.

Reply:
Thanks @erwango, I've addressed this in my latest commit.